client/FxAccountClient.js

```diff
@@ -1012,9 +1023,11 @@ define([
           type: deviceType
         };
 
-        if (options.deviceCallback && options.devicePublicKey) {
+        if (options.deviceCallback && options.devicePublicKey
+            && options.deviceAuthKey) {
```

Comment:
I'm wondering if we need to refactor this. I think there are cases where we update the deviceCallback without the publickey and the deviceauthkey. Seems like the signup above is doing it correctly.